arch/arm/dts/imx6-apalis.dts

```diff
@@ -11,12 +11,6 @@
 	model = "Toradex Apalis iMX6Q/D";
 	compatible = "toradex,apalis_imx6q", "fsl,imx6q";
 
-	/* Will be filled by the bootloader */
-	memory@10000000 {
```

Comment:
If filled by SPL (based on the module version / ram size) then it won't need to be removed, right?

But maybe it can simply just add instead.

One thing to check is how OP-TEE can configure the secure memory regions based on the module revision, as the amount of dram can change from 512MB->2GB iirc. I would imagine SPL to provide an updated device-tree based on the module configuration, but that is not yet done afaik.

Reply:
@ricardosalveti  Well, probably the issue is deeper. This node doesn't need to be in dts as it creates automatically in `fdt_fixup_memory_banks` if not exist. So it is strange why the behavior is different for a pre-claimed node and created in SPL. Investigating.

Reply:
Oh, last I check SPL was indeed updating the node but not replacing it (maybe different syntax), which caused the problem in OP-TEE as it would go and scan every memory node.

If that is indeed the case, then your fix is right and would also be a good target for upstreaming.

Reply:
@MrCry0 any other conclusion from your investigation?

Reply:
It was postponed. Resuming.